test/module/irohad/consensus/yac/yac_proposal_storage_test.cpp

```diff
@@ -42,17 +37,21 @@ TEST_F(YacProposalStorageTest, YacProposalStorageWhenCommitCase) {
       "Init storage => insert unique votes => "
       "expected commit");
 
-  for (auto i = 0u; i < 4; ++i) {
-    ASSERT_EQ(boost::none, storage.insert(valid_votes.at(i)));
-  }
-
-  for (auto i = 4u; i < 7; ++i) {
-    auto commit = storage.insert(valid_votes.at(i));
-    log_->info("Commit: {}", logger::opt_to_string(commit, [](auto answer) {
-                 return "value";
-               }));
-    ASSERT_NE(boost::none, commit);
-    ASSERT_EQ(i + 1, boost::get<CommitMessage>(*commit).votes.size());
+  for (size_t num_inserted = 0; num_inserted < number_of_peers;) {
```

Comment:
Isn't this block similar to?:
https://github.com/hyperledger/iroha/blob/243ba3412d3a1ca5c6dee9d90631660e752a0633/test/module/irohad/consensus/yac/yac_block_storage_test.cpp#L38-L53

Reply:
it is, and you can see that the original tests were quite similar too. Generalizing these would go beyond the scope of this PR.